base/abstractarray.jl

```diff
@@ -905,8 +905,19 @@ julia> getindex(A, 2:4)
  3
  2
  4
+
+ julia> getindex(A, CartesianRange((2:4,)))
+3-element Array{Int64,1}:
+ 3
+ 2
+ 4
 ```
 """
+function getindex{T,N}(A::AbstractArray{T,N}, cr::CartesianRange{CartesianIndex{N}})
+    # transform cartesian range to unit range
+    ur = map((x,y)->x:y, cr.start.I, cr.stop.I)
```

Comment:
The internal representation of `CartesianRange` has changed in 0.7; now it's just a wrapper around an `AbstractUnitRange` tuple, so you can do this with `A[R.indices...]`.

Be sure you do `make testall` before submitting code to base; you'll discover you can't even build this, because `CartesianRange` isn't defined until later in the bootstrap sequence defined by `sysimg.jl`.